### PR TITLE
Adds mpassi-icepack interface for snow flag options

### PR DESCRIPTION
**Pull request by njeffery**
1. Changed 30percent/30percentsw option so that it works like the bulk option in icepack. Suggestions to fix conservation for this option is commented out for further testing but still needed in both column and icepack. (BFB in default option, nonBFB with 30percent or 30percentsw).
2. Added namelist field config_wind_snow_loss_fraction for constant snwlvlfac to pass to icepack (BFB)
3. Corrected snow packages in Registry and config warnings to allow for multiple snow tracer flag options (BFB) 4.Initialization values for snow tracers added to ice_itd.F (nonBFB bugfix, consistent with icepack)
5. Modify the expression for snow ice and snow liquid tracer change due to sublimation - Allows for additional mass loss to come from the snow liquid tracer if the snow ice tracer has a value < rhos.  (nonBFB, consistent with icepack)
6. Adds missing changes to snow ice/liquid tracers from bottom melt. Bugfix. (nonBFB, consistent with icepack)
7. Adds mpas-icepack interface for step-snow in mpas_seaice_icepack.F

### CODE REVIEW

**Comment by njeffery** [comment]
Tested in 1-year single column tests and DTESTS with various snow flags on contrasting column_package and icepack options.  I used the @eclare108213's recent snow updates to icepack for the comparison.   

[Final-snow-mod-comparison-col-vs-icepack.pdf](https://github.com/eclare108213/E3SM/files/12337596/Final-snow-mod-comparison-col-vs-icepack.pdf)

File: components/mpas-seaice/src/Registry.xml [hunk 1]
```diff
@@ -1600,6 +1600,11 @@
 			possible_values="positive real"
 			icepack_name="drhosdwind"
 		/>
+		<nml_option name="config_wind_snow_loss_factor" type="real" default_value="0.3" units="none"
+			description="snow loss factor for wind redistribution"
+			possible_values="positive real less than or equal to 1"
+			icepack_name="snwlvlfac"
```
Comment (eclare108213) [inline components/mpas-seaice/src/Registry.xml:1606]: This description is a little misleading, I think.  The parameter snwlvlfac is the ratio of snow on ridges to that on level ice.  We ran two different kinds of tests originally, first with this being a radiative impact only, having nothing to do with snow loss (moving snow from the level ice to the ridges), and second moving snow from the level ice into the leads, so that there wuold be 30% deeper snow on the ridges than on the level ice.  I think what's coded now is the second use case, so it does affect snow loss to leads, but the 30% factor is the ratio of snow on ridges to that on level ice, and so it's effect as a "snow loss factor" in only indirect.   So I don't think the config label should be `config_wind_snow_loss_factor`.  How about `config_snow_redistribution_factor`?

File: components/mpas-seaice/src/column/ice_colpkg.F90 [hunk 2]
```diff
@@ -2074,19 +2078,24 @@ subroutine colpkg_step_therm1(dt, ncat, nilyr, nslyr, n_aero, &
...
       !---------------------------------------------------------------
       ! 30% rule for snow redistribution: precip factor
       !---------------------------------------------------------------
 
       if (trim(snwredist) == '30percent') then
-         worka = c0      
+         worka = c0
          do n = 1, ncat
-            worka = worka + alvl(n)
+            worka = worka + alvl(n)*aicen(n)
```
Comment (eclare108213) [inline components/mpas-seaice/src/column/ice_colpkg.F90:2091]: I questioned whether this is right, because the 30% ratio is relevant only for the ice area, not the full grid cell area (which is what multiplication by aicen expands to).  Our write-up from ~8 years ago (section 4.1.1, attached) says "The factor of ai is included implicitly, as the snowfall flux is applied only over the ice area."  That's true in general (regardless of whether snow redistribution is on or off) but I think it isn't the full story here. A more fundamental error in the original parameterization is that the calculations in the document are for one category (they are missing the summations over multiple categories), and so ai can not be divided out so cleanly as shown there or assumed in the original code.  This change fixes that problem.
[snowdoc.pdf](https://github.com/eclare108213/E3SM/files/12383803/snowdoc.pdf)

File: components/mpas-seaice/src/column/ice_colpkg.F90 [hunk 3]
```diff
@@ -2074,19 +2078,24 @@ subroutine colpkg_step_therm1(dt, ncat, nilyr, nslyr, n_aero, &
...
          enddo
-         worka  = worka * snwlvlfac/(c1+snwlvlfac)
-         fsloss = fsloss + fsnow*(c1-worka)
-         fsnow  =          fsnow*    worka
+         worka  = worka * snwlvlfac/(c1+snwlvlfac)/aice
+         fsloss = fsloss + fsnow * worka
```
Comment (eclare108213) [inline components/mpas-seaice/src/column/ice_colpkg.F90:2094]: The new fsloss calculation here looks correct, based on Delta V in the snowdoc.pdf.  Note that the *worka and *(1-worka) factors are reversed now, and this has also been changed in Icepack.

**Review by eclare108213 — COMMENTED** [review]
I've convinced myself that the changes here are correct.  The 'bulk' 30% differences from the originally worried me at first, but I think I understand them (and made comments with my conclusions, for future reference).  I have not checked the conservation aspects, which are commented out here. 
Before approving this PR I want to update Icepack from the Consortium (with related snow changes) and retest all of it.  Looks good, though.  Thanks!

**Comment by eclare108213** [comment]
@dabail10 in case you're still hunting for the bulk wind redistribution changes for conservation, they are commented out in this PR.
